awscli/customizations/s3/subcommands.py

```diff
@@ -985,11 +985,18 @@ def run(self):
             'result_queue': result_queue,
         }
 
-        fgen_request_parameters = {}
+        fgen_request_parameters = {
+            'RequestPayer': self.parameters.get('request_payer'),
+        }
         fgen_head_object_params = {}
         fgen_request_parameters['HeadObject'] = fgen_head_object_params
         fgen_kwargs['request_parameters'] = fgen_request_parameters
 
+        rgen_request_parameters = {
```

Comment:
One question that I have is for a cross-copy `sync` command that has one bucket that uses requester pays and the other one that does not use requester pays, is there any unintentional side effects of passing `RequestPayer` to a `ListObjects` (or any other) call for the bucket where request payer is not enabled. I do not think so but I would need to double check this. Just wanted to see if you ran into this at all.